src/SprykerEco/Yves/Payone/Handler/PayoneHandler.php

```diff
@@ -157,6 +159,12 @@ protected function setPayonePayment(Request $request, QuoteTransfer $quoteTransf
             $paymentDetailTransfer->setBankGroupType($payonePaymentTransfer->getBankGroupType());
             $paymentDetailTransfer->setIban($payonePaymentTransfer->getIban());
             $paymentDetailTransfer->setBic($payonePaymentTransfer->getBic());
+        } elseif ($paymentSelection == PaymentTransfer::PAYONE_CASH_ON_DELIVERY) {
```

Comment:
This is worst place of the module... It must be refactored, but maybe not in this PR.

Reply:
Agree, but such a refactoring is out of the scope